### ISSUE TITLE
Not able to retrieve all values on streaming services

### ISSUE THREAD

**Issue by nlagaros** [SoCo/SoCo#547]
I think think is particular to streaming services.  I've tested on Pandora and Spotify.  I can only retrieve the station name and track - not the creator/artist nor album. See the errors below in the debug log: "Update no possible, no resources".  I'm not sure this is the cause -- but it seems like all the content is in the raw event.

````
INFO:soco.events_base_twisted:Event 9 received for AVTransport service at 1511613207.81
DEBUG:soco.events:Event content: <e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0"><e:property><LastChange>&lt;Event xmlns=&quot;urn:schemas-upnp-org:metadata-1-0/AVT/&quot; xmlns:r=&quot;urn:schemas-rinconnetworks-com:metadata-1-0/&quot;&gt;&lt;InstanceID val=&quot;0&quot;&gt;&lt;TransportState val=&quot;PLAYING&quot;/&gt;&lt;CurrentPlayMode val=&quot;NORMAL&quot;/&gt;&lt;CurrentCrossfadeMode val=&quot;0&quot;/&gt;&lt;NumberOfTracks val=&quot;2&quot;/&gt;&lt;CurrentTrack val=&quot;1&quot;/&gt;&lt;CurrentSection val=&quot;0&quot;/&gt;&lt;CurrentTrackURI val=&quot;x-sonos-http:PIkfvTQeQmuOQyWuYqXrZdf216378HuQeu2hm1neH4gCu00rvGNq1BRh7P5DPph_P3o91Dzbn9sFKZ5fyKCg7Lw%3a%3aST%3a3708522527768206320%3a%3aRINCON_B8E93770DE6B01400%3a1601.mp3?sid=236&amp;amp;flags=0&amp;amp;sn=10&quot;/&gt;&lt;CurrentTrackDuration val=&quot;0:03:20&quot;/&gt;&lt;CurrentTrackMetaData val=&quot;&amp;lt;DIDL-Lite xmlns:dc=&amp;quot;http://purl.org/dc/elements/1.1/&amp;quot; xmlns:upnp=&amp;quot;urn:schemas-upnp-org:metadata-1-0/upnp/&amp;quot; xmlns:r=&amp;quot;urn:schemas-rinconnetworks-com:metadata-1-0/&amp;quot; xmlns=&amp;quot;urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/&amp;quot;&amp;gt;&amp;lt;item id=&amp;quot;-1&amp;quot; parentID=&amp;quot;-1&amp;quot;&amp;gt;&amp;lt;res protocolInfo=&amp;quot;http-get:*:audio/mp3:*&amp;quot; duration=&amp;quot;0:03:20&amp;quot;&amp;gt;x-sonos-http:PIkfvTQeQmuOQyWuYqXrZdf216378HuQeu2hm1neH4gCu00rvGNq1BRh7P5DPph_P3o91Dzbn9sFKZ5fyKCg7Lw%3a%3aST%3a3708522527768206320%3a%3aRINCON_B8E93770DE6B01400%3a1601.mp3?sid=236&amp;amp;amp;flags=0&amp;amp;amp;sn=10&amp;lt;/res&amp;gt;&amp;lt;upnp:albumArtURI&amp;gt;https://mediaserver-cont-ch1-1-v4v6.pandora.com/images/public/rovi/albumart/7/2/7/4/008811194727_500W_500H.jpg&amp;lt;/upnp:albumArtURI&amp;gt;&amp;lt;upnp:class&amp;gt;object.item.audioItem.musicTrack&amp;lt;/upnp:class&amp;gt;&amp;lt;dc:title&amp;gt;Sweet Caroline&amp;lt;/dc:title&amp;gt;&amp;lt;dc:creator&amp;gt;Neil Diamond&amp;lt;/dc:creator&amp;gt;&amp;lt;r:albumArtist&amp;gt;Neil Diamond&amp;lt;/r:albumArtist&amp;gt;&amp;lt;upnp:album&amp;gt;The Best Of Neil Diamond - 20th Century Masters: The Millennium Collection&amp;lt;/upnp:album&amp;gt;&amp;lt;r:trackGain&amp;gt;-4.480000&amp;lt;/r:trackGain&amp;gt;&amp;lt;r:tiid&amp;gt;PIkfvTQeQmuOQyWuYqXrZdf216378HuQeu2hm1neH4gCu00rvGNq1BRh7P5DPph_P3o91Dzbn9sFKZ5fyKCg7Lw&amp;lt;/r:tiid&amp;gt;&amp;lt;r:rating&amp;gt;&amp;lt;r:type&amp;gt;THUMBSUP&amp;lt;/r:type&amp;gt;&amp;lt;r:connotation&amp;gt;POSITIVE&amp;lt;/r:connotation&amp;gt;&amp;lt;/r:rating&amp;gt;&amp;lt;r:policies mask=&amp;quot;0x80020008&amp;quot; flags=&amp;quot;0x20008&amp;quot;/&amp;gt;&amp;lt;/item&amp;gt;&amp;lt;/DIDL-Lite&amp;gt;&quot;/&gt;&lt;r:NextTrackURI val=&quot;&quot;/&gt;&lt;r:NextTrackMetaData val=&quot;&quot;/&gt;&lt;r:EnqueuedTransportURI val=&quot;x-sonosapi-radio:ST%3a3708522527768206320?sid=236&amp;amp;flags=0&amp;amp;sn=10&quot;/&gt;&lt;r:EnqueuedTransportURIMetaData val=&quot;&amp;lt;DIDL-Lite xmlns:dc=&amp;quot;http://purl.org/dc/elements/1.1/&amp;quot; xmlns:upnp=&amp;quot;urn:schemas-upnp-org:metadata-1-0/upnp/&amp;quot; xmlns:r=&amp;quot;urn:schemas-rinconnetworks-com:metadata-1-0/&amp;quot; xmlns=&amp;quot;urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/&amp;quot;&amp;gt;&amp;lt;item id=&amp;quot;000c0000ST%3a3708522527768206320&amp;quot; parentID=&amp;quot;-1&amp;quot; restricted=&amp;quot;true&amp;quot;&amp;gt;&amp;lt;dc:title&amp;gt;The Beatles Radio&amp;lt;/dc:title&amp;gt;&amp;lt;upnp:class&amp;gt;object.item.audioItem.audioBroadcast&amp;lt;/upnp:class&amp;gt;&amp;lt;desc id=&amp;quot;cdudn&amp;quot; nameSpace=&amp;quot;urn:schemas-rinconnetworks-com:metadata-1-0/&amp;quot;&amp;gt;SA_RINCON60423_X_#Svc60423-dfc5b1b3-Token&amp;lt;/desc&amp;gt;&amp;lt;upnp:albumArtURI&amp;gt;https://mediaserver-cont-sv5-1-v4v6.pandora.com/images/public/gracenote/albumart/4/4/9/4/4988031174944_1080W_1080H.jpg&amp;lt;/upnp:albumArtURI&amp;gt;&amp;lt;/item&amp;gt;&amp;lt;/DIDL-Lite&amp;gt;&quot;/&gt;&lt;/InstanceID&gt;&lt;/Event&gt;</LastChange></e:property></e:propertyset>
DEBUG:soco.services:Cache hit
DEBUG:soco.music_services.accounts:Fetching account data from <SoCo object at ip 192.168.0.61>
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): 192.168.0.61
DEBUG:urllib3.connectionpool:http://192.168.0.61:1400 "GET /status/accounts HTTP/1.1" 200 None
DEBUG:soco.music_services.accounts:Account data: <?xml version="1.0" ?>
<?xml-stylesheet type="text/xsl" href="/xml/review.xsl"?><ZPSupportInfo><Accounts LastUpdateDevice="RINCON_B8E93770DE6B01400" Version="563" NextSerialNum="14"><Account Type="60423" SerialNum="2" Flags="0"><UN>X_#Svc60423-4189a4c1-Token</UN><MD>1</MD><NN>Nicholas</NN><OADevID>HHID_x85Mp7oIPAj3rO9ZzezQBKdZTXI_4189a4c1</OADevID><Hash>GQe+YaVNDi9JeQ4JH+3OUw==</Hash></Account><Account Type="3079" SerialNum="4" Flags="0"><UN>X_#Svc3079-0-Token</UN><MD>1</MD><NN>spartacus1896</NN><OADevID>HHID_x85Mp7oIPAj3rO9ZzezQBKdZTXI</OADevID><Hash>1qoMkIXMOEqm48f9MhAWpQ==</Hash></Account><Account Type="51463" SerialNum="9" Flags="0"><UN>X_#Svc51463-0-Token</UN><MD>1</MD><NN>Marie</NN><OADevID>HHID_x85Mp7oIPAj3rO9ZzezQBKdZTXI</OADevID><Hash></Hash></Account><Account Type="60423" SerialNum="10" Flags="0"><UN>X_#Svc60423-dfc5b1b3-Token</UN><MD>1</MD><NN>Marie</NN><OADevID>HHID_x85Mp7oIPAj3rO9ZzezQBKdZTXI_dfc5b1b3</OADevID><Hash>u2oNv+7H1o9jJljEw7uPBw==</Hash></Account><Account Type="52231" SerialNum="12" Flags="0"><UN>X_#Svc52231-0-Token</UN><MD>1</MD><NN>Apple ID</NN><OADevID>HHID_x85Mp7oIPAj3rO9ZzezQBKdZTXI</OADevID><Hash></Hash></Account><Account Type="9479" SerialNum="13" Flags="0"><UN>nicholas@lagaros.com</UN><MD>1</MD><NN>Nicholas</NN><OADevID></OADevID><Hash></Hash></Account><Account Type="9479" SerialNum="8" Flags="0" Deleted="1"><UN>5163167505</UN><MD></MD><NN>nicholas</NN><OADevID></OADevID><Hash></Hash></Account><Account Type="1543" SerialNum="1" Flags="0" Deleted="1"><UN>Sonos</UN><MD></MD><NN></NN><OADevID></OADevID><Hash></Hash></Account><Account Type="4359" SerialNum="6" Flags="0" Deleted="1"><UN>nicholas@lagaros.com</UN><MD></MD><NN>nicholas</NN><OADevID></OADevID><Hash></Hash></Account><Account Type="7431" SerialNum="3" Flags="0" Deleted="1"><UN>nlagaros</UN><MD></MD><NN></NN><OADevID></OADevID><Hash></Hash></Account><Account Type="6663" SerialNum="11" Flags="0" Deleted="1"><UN></UN><MD></MD><NN></NN><OADevID></OADevID><Hash></Hash></Account></Accounts></ZPSupportInfo>
DEBUG:soco.music_services.data_structures:MSTrack.__init__ with item_id=11111111PIkfvTQeQmuOQyWuYqXrZdf216378HuQeu2hm1neH4gCu00rvGNq1BRh7P5DPph_P3o91Dzbn9sFKZ5fyKCg7Lw%3a%3aST%3a3708522527768206320%3a%3aRINCON_B8E93770DE6B01400%3a1601, desc=SA_RINCON60423_X_#Svc60423-dfc5b1b3-Token, resources=[<DidlResource 'x-sonos-http:PIkfvTQeQmuOQyWuYqXrZdf216378HuQeu2hm1neH4gCu00rvGNq1BRh7P5DPph_P3o91Dzbn9sFKZ5fyKCg7Lw%3a%3aST%3a3708522527768206320%3a%3aRINCON_B8E93770DE6B01400%3a1601.mp3?sid=236&flags=0&sn=10' at 0x10927b350>], uri=x-sonos-http:PIkfvTQeQmuOQyWuYqXrZdf216378HuQeu2hm1neH4gCu00rvGNq1BRh7P5DPph_P3o91Dzbn9sFKZ5fyKCg7Lw%3a%3aST%3a3708522527768206320%3a%3aRINCON_B8E93770DE6B01400%3a1601.mp3?sid=236&flags=0&sn=10, metadata_dict=..., music_service=None
DEBUG:soco.music_services.data_structures:MetadataDictBase.__init__ with: {'title': u'Sweet Caroline'}
DEBUG:soco.data_structures_entry:Item <DidlMusicTrack 'Sweet Caroline' at 0x10927bf10> upgraded to <MSTrack title="Sweet Caroline">
DEBUG:soco.data_structures_entry:Created data structures: [<soco.music_service (CUT) from Didl string "<DIDL-Lite xmlns:dc=" (CUT)
DEBUG:soco.data_structures_entry:Upgrade not possible, no resources
DEBUG:soco.data_structures_entry:Created data structures: [<DidlAudioBroadcast (CUT) from Didl string "<DIDL-Lite xmlns:dc=" (CUT)
'AVTransport'
{u'current_crossfade_mode': '0',
 u'current_play_mode': 'NORMAL',
 u'current_section': '0',
 u'current_track': '1',
 u'current_track_duration': '0:03:20',
 u'current_track_meta_data': <soco.music_services.data_structures.MSTrack object at 0x1092924d0>,
 u'current_track_uri': 'x-sonos-http:PIkfvTQeQmuOQyWuYqXrZdf216378HuQeu2hm1neH4gCu00rvGNq1BRh7P5DPph_P3o91Dzbn9sFKZ5fyKCg7Lw%3a%3aST%3a3708522527768206320%3a%3aRINCON_B8E93770DE6B01400%3a1601.mp3?sid=236&flags=0&sn=10',
 u'enqueued_transport_uri': 'x-sonosapi-radio:ST%3a3708522527768206320?sid=236&flags=0&sn=10',
 u'enqueued_transport_uri_meta_data': <DidlAudioBroadcast 'The Beatles Radio' at 0x109292110>,
 u'next_track_meta_data': '',
 u'next_track_uri': '',
 u'number_of_tracks': '2',
 u'transport_state': 'PLAYING'}
```
`

**Comment by ghcs27** [comment]
I'd suspect the information gets lost in the upgrade process, as observed and quick-patched in #559.

**Comment by KennethNielsen** [comment]
I think should be fixed now via #589, if that is not the case feel free to re-open
